internal/venice-common/src/main/java/com/linkedin/venice/system/store/MetaStoreWriter.java

```diff
@@ -371,16 +372,31 @@ private void update(
     }
     StoreMetaKey key = dataType.getStoreMetaKey(keyStringSupplier.get());
     SpecificRecord update = updateSupplier.get();
-    writer.update(
-        key,
-        update,
-        AvroProtocolDefinition.METADATA_SYSTEM_SCHEMA_STORE.currentProtocolVersion.get(),
-        derivedComputeSchemaId,
-        null);
-    writer.flush();
+    ReentrantLock lock = getOrCreateMetaStoreWriterLock(metaStoreName);
+    VeniceWriter writer = null;
+    try {
+      lock.lock();
+      writer = getOrCreateMetaStoreWriter(metaStoreName);
+      writer.update(
+          key,
+          update,
+          AvroProtocolDefinition.METADATA_SYSTEM_SCHEMA_STORE.currentProtocolVersion.get(),
+          derivedComputeSchemaId,
+          null);
+      writer.flush();
+    } catch (Exception e) {
+      LOGGER.error("Caught exception while trying to update, will respawn a new writer.", e);
+      closeVeniceWriter(metaStoreName, writer, true);
```

Comment:
writer could be null here ?

Reply:
getOrCreateMetaStoreWriter should always return a VW is my understanding here.